shareprogress/services.py

```diff
@@ -1,64 +1,107 @@
+import json
+
 # Managing Buttons
-def create_button(page_url, button_template, page_title=None, *args):
-    """Creates a ShareProgress button and returns a JSON object as response.
+def create_button(page_url, button_template, variants, page_title=None):
+    """Creates a JSON object for a share button to be sent to ShareProgress.
 
     Arguments received
 
     page_url: The URL of the page to be shared.
-    
+
     button_template: By knowing the button_template we know which channel
     is going to be set for that button (email, twitter or facebook).
-    
-    page_title: It's optional, when 'None' it will be scraped from the 
-    page_url automatically.
-    
-    *args: Received arguments for the variant/s
+
+    variants: Received arguments for the variant/s
     we would use the create_variant() function here.
-    
-    It will return a JSON object with the ID of the button created and
-    its information (included the HTML tag for the button to be embedded).
+
+    page_title: It's optional, when set to 'None' it will be scraped from the
+    page_url automatically.
     """
 
-def update_button(id, *args):
+    body = {}
+
+    body['key'] = '123456'      # This will be taken from .env
+    body['page_url'] = page_url
+
+    if (page_title is not None):
+        body['page_title'] = page_title
+
+    # button_template: i.e. 'sp_em_small' or 'sp_tw_large'
+    body['button_template'] = button_template
+
+    body['auto_fill'] = True    # Allows page_url scraping to fill info in
+
+    body['variants'] = {}
+
+    if (button_template[3:5] == 'em'):
+        body['variants']['email'] = []
+
+        for v in variants:
+            email_variant = create_variant('email', v)
+
```

Comment:
I'd drop the extra line breaks in places like this. They're not breaking any style rules, but I don't think the extra line break here and elsewhere makes the code more readable.